test/Constraints/members.swift

```diff
@@ -664,3 +664,12 @@ func test_34770265(_ dict: [Int: Int]) {
   dict.rdar_34770265_val()
   // expected-error@-1 {{referencing instance method 'rdar_34770265_val()' on 'Dictionary' requires the types 'Int' and 'String' be equivalent}}
 }
+
+// SR-12672
+_ = [.e] // expected-error {{reference to member 'e' cannot be resolved without a contextual type}}
+let _ : [Any] = [.e] // expected-error {{type 'Any' has no member 'e'}}
+_ = [1 :.e] // expected-error {{reference to member 'e' cannot be resolved without a contextual type}}
+_ = [.e: 1] // expected-error {{reference to member 'e' cannot be resolved without a contextual type}}
+let _ : [Int: Any] = [1 : .e] // expected-error {{reference to member 'e' cannot be resolved without a contextual type}}
```

Comment:
This seems not right, but I have no idea why there is no contextual type being returned here ... @xedin

Reply:
See my comment below, it applies to contextual type check :)

Reply:
Thank's @xedin, I've just updated :)